src/screens/AddNewTerm.js

```diff
@@ -0,0 +1,193 @@
+import React, {useRef} from 'react';
+import {action} from '@storybook/addon-actions';
+
+import {
+  View,
+  SafeAreaView,
+  StyleSheet,
+  KeyboardAvoidingView,
+} from 'react-native';
+import SelectDropdown from 'react-native-select-dropdown';
+
+import {LANGUAGE_NAMES} from '../data/dataUtils';
+import Textarea from '../components/Textarea/Textarea';
+import ToolBar from '../components/ToolBar/ToolBar';
+import ToolButton from '../components/ToolButton/ToolButton';
+import NextButton from '../components/NextButton/NextButton';
+import SectionHeading from '../components/SectionHeading/SectionHeading';
+import LanguageSwitcher from '../components/LanguageSwitcher/LanguageSwitcher';
+
+import BackIcon from '../components/ToolButton/assets/back.svg';
+import ModeIcon from '../components/ToolButton/assets/mode.svg';
+import ArrowDownIcon from '../icons/arrow-down.svg';
+import {KeyboardAwareScrollView} from '@codler/react-native-keyboard-aware-scroll-view';
+
+export default ({
+  // Nav provider
+  navigation,
+  categories,
+  englishTerm,
+  malagasyTerm,
+  setEnglishTerm,
+  setMalagasyTerm,
+  selectedCategory,
+  setSelectedCategory,
+  nativeLanguage,
+  addUserPhrase,
+}) => {
+  // Dropdown selector: reset and handle default text
+  const dropdownRef = useRef({});
+  const defaultTextSelecter = 'Select Category';
+  const IconWrapper = () => {
+    return (
+      <View style={{marginRight: 115}}>
+        <ArrowDownIcon />
+      </View>
+    );
+  };
+
+  function englishWordInput(e) {
+    setEnglishTerm(e);
+  }
+
+  function malagasyWordInput(e) {
+    setMalagasyTerm(e);
+  }
```

Comment:
instead of the above use useState. EnglishTerm and MalagasyTerm dont have to be in redux state as no other components are using those values

Reply:
I have made the change and instead of creating a function I uses (e) => setState(e) which makes the code smaller.

Thanks.